gnucash/gnucash-bin.c

```diff
@@ -600,6 +610,72 @@ get_file_to_load()
         return gnc_history_get_last();
 }
 
+static void
+inner_main_run_report(void *closure, int argc, char **argv)
+{
+    SCM scm_report_output;
+    QofSession *session = NULL;
+
+    scm_c_eval_string("(debug-set! stack 200000)");
+    scm_set_current_module (scm_c_resolve_module ("gnucash utilities"));
+    scm_c_use_module("gnucash app-utils");
+
+    scm_c_use_module("gnucash reports");
+    scm_c_use_module("gnucash report-menus");
+
+    /* scm_set_current_module (scm_c_resolve_module ("gnucash report")); */
+    /* scm_set_current_module (scm_c_resolve_module ("gnucash reports")); */
+    gnucash_register_add_cell_types ();
+    gnc_report_init ();
+    load_gnucash_plugins();
+    load_gnucash_modules();
+    load_system_config();
+    load_user_config();
+    /* gnc_plugin_report_system_new(); */
+    gnc_hook_run(HOOK_STARTUP, NULL);
+
+    gnc_prefs_init ();
+    qof_event_suspend ();
+
+    printf ("Loading datafile...\n");
+    session = gnc_get_current_session ();
+    if (!session) goto fail;
+
+    qof_session_begin (session, get_file_to_load(), FALSE, FALSE, FALSE);
+    if (qof_session_get_error (session) != ERR_BACKEND_NO_ERR) goto fail;
+
+    qof_session_load (session, NULL);
+    if (qof_session_get_error (session) != ERR_BACKEND_NO_ERR) goto fail;
+
+    if (report_output)
+        scm_report_output = scm_from_utf8_string (report_output);
+    else
+        scm_report_output = scm_from_utf8_string ("gnucash.html");
```

Comment:
If you would consider outputting to stdout it may be easier to keep output file handling in C(++) and reduce guile's responsibility to generate the html to a string.